mediatek: change Routerich AX3000 ubi size

romanovj opened the pull request:
All new routers are shipped with ubi size 112MB since early September.

Bootloader update required (ask vendor , see wiki)

These partitions weren't used:
firmware_backup
zrsave
config2

MisharUser reviewed (commented):
I checked the work of this commit, the firmware is working, and it took up all the space, I confirm that this change will benefit

hauke commented:
This prevents an update from devices with the old flash size to this new version. We already have support for this device with the old flash size in OpenWrt 23.05.

What was happening there?
Did the vendor change the partition layout for new devices to a bigger UBI partition?
Is this only a change some people are doing in an OpenWrt fork or a fork of the vendor firmware?
Does the vendor also apply this change to older devices which shipped with the smaller UBI partition size before in a normal firmware upgrade?

We can not do a normal upgrade from the old size to the new one you have to at least change the DEVICE_COMPAT_VERSION.
Maybe it even makes sense to ship two different images one for the old one and one with the new size.

@csharper2005 what do you think?

csharper2005 commented:
@hauke I consider the option with two types of images to be unnecessary, since there is no OpenWrt U-Boot for either option. I like the option with DEVICE_COMPAT_VERSION + short reference to the OpenWrt wiki + clear guide on WiKi with direct links to the new FIP and preloader.

@romanovj we should use DEVICE_COMPAT_VERSION to avoid bricking and configuration loss. Сould you show new U-Boot bootlog, please? Is it NBMB still enabled? If not then we need additional changes.

csharper2005 commented:
> @csharper2005 the same uboot but with
> 
> `CONFIG_MTDPARTS_DEFAULT="nmbm0:1024k(bl2),512k(u-boot-env),2048k(Factory),2048k(fip),112M(ubi)" `

Can you share real bootlog, please?

csharper2005 commented:
@romanovj thanks!

I wonder why the vendor didn't extend the UBI to 0x7800000...

csharper2005 commented:
> > @romanovj thanks!
> > I wonder why the vendor didn't extend the UBI to 0x7800000...
> 
> and fip + eeprom - 4MB instead of ~1MB

I think this PR should be merged after taking into account the reviewers suggestions above. And we should think about OpenWrt U-Boot layout variant. It allows:
- Avoid inconveniences for official OpenWrt users when changes are made on the vendor's side (like now);
- Get at least 16 MiB of additional space without NMBM (compared to the layout proposed in this PR).

erantake committed:
mediatek: change Routerich AX3000 ubi size
All new routers are shipped with ubi size 112MB since early September.
Bootloader update required (ask vendor , see wiki)

These partitions weren't used:
firmware_backup
zrsave
config2

Signed-off-by: Romanov Danila <pervokur@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16686
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

csharper2005 commented:
> @csharper2005 can you check it?

The code looks good to me. Is it runtested?

> https://openwrt.org/toh/routerich/ax3000#web_ui_method

Warning will show this link. There is no links to the updated fip/bl2 there. How will the user get them?

csharper2005 commented:
> "contact your vendor and follow the guide:" Users will contact vendor and get files and guides in Russian.

This situation is a bit strange to me. We need third-party (vendor) to upgrade official OpenWrt 23.05.x to official OpenWrt snapshot after this commit get merged. This is an unnecessary dependency.

csharper2005 commented:
> @csharper2005 I can add link to post from the routerich telegram channel with ubi 64M/ubi 112M uboots and md5 c-sums. What do you think would be best?

That's possible. I opened a PR for OpenWrt U-Boot layout https://github.com/openwrt/openwrt/pull/16791. I will update it if your PR get merged sooner.

RomanovDanila commented:
@dangowrt , can you merge this code so that everyone will have the same layout?
